### PR TITLE
Adding MacOS E2E Testing to CI Pipeline

### PR DESCRIPTION
**Pull request by samuelfreiberg**
### Platforms Impacted
- [ ] iOS
- [X] macOS
- [ ] win32 (Office)
- [ ] windows
- [ ] android

### Description of changes

Adding MacOS E2E Testing to CI Pipeline

### Verification

(how the change was tested, including both manual and automated tests)

| Before                                       | After                                      |
|----------------------------------------------|--------------------------------------------|
| Screenshot or description before this change | Screenshot or description with this change |

### Pull request checklist

This PR has considered (when applicable):
- [ ] Automated Tests
- [ ] Documentation and examples
- [ ] Keyboard Accessibility
- [ ] Voiceover
- [ ] Internationalization and Right-to-left Layouts

### CODE REVIEW

**Comment by Saadnajmi** [comment]
Would like to hold this one till #1535 merges. Just had our sync and I'll try to get that merged ASAP, aka this week.

**Comment by samuelfreiberg** [comment]
> Would like to hold this one till #1535 merges. Just had our sync and I'll try to get that merged ASAP, aka this week.

Definitely, was planning on waiting. Just wanted to test the CI to make sure we only run this for MacOS and not iOS. The matrix part of the pipeline is a bit confusing to me, but I'm working on it :)

File: .ado/azure-pipelines.yml [hunk 1]
```diff
@@ -116,6 +125,10 @@ jobs:
         parameters:
           relative_directory: $(relative_directory)
 
+      # - ${{ if eq(variables[relative_directory], 'macos')}}:
```
Comment (harrieshin) [inline .ado/azure-pipelines.yml:128]: ?

File: .ado/azure-pipelines.yml [hunk 2]
```diff
@@ -93,6 +93,15 @@ jobs:
       - checkout: self
         persistCredentials: true
 
+      - bash: echo Test 1
+        condition: and(succeeded(), eq($(relative_directory), 'macos'))
+
+      - bash: echo Test 2
+        condition: and(succeeded(), eq($(relative_directory), macos))
+
+      - bash: echo Test 3
```
Comment (harrieshin) [inline .ado/azure-pipelines.yml:102]: what are we testing here?

File: .ado/azure-pipelines.yml (hunk 1 above)
Comment (Saadnajmi) [reply .ado/azure-pipelines.yml:128]: YML is hard to verify. locally so most likely you'll see a lot of small commits to try and debug the yml till it works

File: .ado/azure-pipelines.yml (hunk 2 above)
Comment (samuelfreiberg) [reply .ado/azure-pipelines.yml:98]: I'm going to remove this code, I just needed to test what syntax would work for certain conditions. I'm going to mark this PR as a draft until I figure it out :)

File: .ado/azure-pipelines.yml (hunk 1 above)
Comment (samuelfreiberg) [reply .ado/azure-pipelines.yml:124]: Testing. Will mark this PR as a draft until I get it down

**Pull request closed without merging**